src/PETScLinearSolvers.jl

```diff
@@ -0,0 +1,107 @@
+
+struct PETScLinearSolver{F} <: LinearSolver
+  setup::F
+  comm::MPI.Comm
```

Comment:
Yes, I was aware (even before the comment) that we were pursuing this. 

However, I tried, and I do not see how this can be achieved cleanly. 

Currently, the communicator is passed through symbolic setup and then numerical setup, which is the point at which it is needed. 

If we remove comm from PETScLinearSolver, we have to write several specializations of `numerical_setup` distpatching on the type of input matrix to factorize, so that we can extract the communicator frrm it. Is that what you have in mind? If yes, I can proceed.

Reply:
I have realized that the following definition 

```
PETScLinearSolver() = PETScLinearSolver(MPI.COMM_WORLD)
```

is dangerous. I think we should always provide the actual communicator or extract the communicator from an actual distributed object (see me previous comment).

Reply:
> If we remove comm from PETScLinearSolver, we have to write several specializations of numerical_setup

Perhaps it is not needed to have several specialisations. We can just get the comm after converting to petsc matrix
```julia
function Algebra.numerical_setup(ss::PETScLinearSolverSS,A::AbstractMatrix)
  B = convert(PETScMatrix,A)
  comm = B.comm
  # etc.
```

> I have realized that the following definition

Yes I agree, this is VERY dangerous! Another indicator that we need to remove `comm` from `PETScLinearSolver`.

Reply:
Ok, I think this is already solved. Ready to merge (if passing tests) ... Should we also touch the PETSc version in Project.toml?

Reply:
> Should we also touch the PETSc version in Project.toml?

Yes, Done!